### ISSUE TITLE
Can't run truffle migrate with --light --testnet

### ISSUE THREAD

**Issue by HaleTom** [ethereum/go-ethereum#15366]
**Note:** [This truffle issue](https://github.com/trufflesuite/truffle/issues/534) asserts that the problem lies in using `geth --light`, so that is why I'm raising it here.

#### System information

Geth version: `Version: 1.7.1-stable`
OS & Version: `Linux svelte 4.9.52-1-MANJARO #1 SMP PREEMPT Wed Sep 27 16:40:38 UTC 2017 x86_64 GNU/Linux`

#### Expected behaviour
I can run `truffle migrate --reset` when running `geth --light --testnet` like I can when running with `testrpc`.

```
$ truffle migrate --reset                      
Using network 'development'.                   

Running migration: 1_initial_migration.js      
  Deploying Migrations...                      
  ... 0xba60d5cc8038d2013b5c72fe030b3e5713a2c64852cb26689065a9b221363445                       
  Migrations: 0xd2c52206f981288e4e2fee9edcddae8601bedac9                                       
Saving successful migration to network...      
  ... 0x526e663bcc8099a75b397275bb940f2557dd27792fd4c5e5a4ef2617ed1d4f3f                       
Saving artifacts...                            
Running migration: 2_deploy_contracts.js       
  Deploying TutorialToken...                   
  ... 0x2958de349a3acba76bd94c510129fdd0d9fe72c6587249e40b39244d009194b0                       
  TutorialToken: 0x62b2a91ad38b864a9e7488c7881d353fa5b1673e                                    
Saving successful migration to network...      
  ... 0x0362d247dd7ea4f01df52a62ba217d252d2e27ef61c41b872c7fd71af469dce6                       
Saving artifacts...
```

#### Actual behaviour
```
$ truffle migrate --reset --network test       
Using network 'test'.                          

Running migration: 1_initial_migration.js      
  Deploying Migrations...                      
  ... 0x53c00e4aed7d9a3ec63d2583c0bb5db78b8e783fac5c027f9e1c5573d7ba0977                       
/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:30432                
    throw new Error('invalid address');        
    ^                                          

Error: invalid address                         
    at inputAddressFormatter (/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:30432:11)                                                                             
    at /home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:35007:28      
    at Array.map (native)                      
    at Method.formatInput (/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:35006:32)                                                                                
    at Method.toPayload (/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:35032:23)                                                                                  
    at Eth.send [as getCode] (/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:35057:30)                                                                             
    at Object.callback (/home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:314845:39)
    at /home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:35060:25      
    at /home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:316808:9      
    at /home/ravi/.local/share/npm/lib/node_modules/truffle/build/cli.bundled.js:164746:11     
```

#### Steps to reproduce the behaviour
Run `truffle migrate` pointing to the RPC port of `geth --light --testnet`

#### Other information
See [this truffle issue](https://github.com/trufflesuite/truffle/issues/534)'s  investigation which purportedly shows that the issue lies in using `geth --light`.

**Comment by fabdarice** [comment]
+1
Anybody from geth or truffle could confirm if it's possible to deploy a contract with geth running in light mode ?

**Comment by ivica7** [comment]
I would also like to know why this is not possible. Which information is missing in light-mode?

**Comment by fabdarice** [comment]
@ivica7 Hi, after investigation, the contract deploy correctly in light mode.
HOWEVER, it doesn't render correctly a contractAddress after calling web3.eth.getTransactionReceipt, which is causing Truffle to fail during the deployment.

I'm working on a PR to fix this issue in truffle with a work-around. Will keep you posted.

**Comment by fabdarice** [comment]
@ivica7 issue has been fixed with geth 1.8.2
